Document Bookmarks & Reminders (#333)
discord-userdoccers/discord-userdoccers: pages/resources/user.mdx

| channel_id | snowflake | The ID of the channel the message is in               |
| message_id | snowflake | The ID of the message to apply the confetti potion to |
| emoji_name | string    | Unicode emoji or custom emoji name and ID             |

<RouteHeader method="GET" url="/users/@me/saved-messages">
  Get Saved Messages
</RouteHeader>

> Returns the list of saved messages for the current user.

Reviewer: ```suggestion
Returns message bookmarks and reminders for the current user.
```

| --------- | ---------------------------------------------------- | ----------------------------- |
| message   | ?[message](/resources/message#message-object) object | The saved message             |
| save_data | [save data](#save-data-structure) object             | The save data for the message |

###### Save Data Structure

| Field           | Type               | Description                              |
| --------------- | ------------------ | ---------------------------------------- |
| channel_id      | snowflake          | The ID of the channel                    |
| message_id      | snowflake          | The ID of the message                    |
| guild_id?       | snowflake          | The ID of the guild                      |
| saved_at        | ISO8601 timestamp  | The timestamp when the message was saved |
| author_summary  | string             | Unknown                                  |
| channel_summary | string             | Unknown                                  |
| message_summary | string             | Unknown                                  |
| notes           | string             | Unknown                                  |

> | due_at          | ?ISO8601 timestamp | Due timestamp for the reminder           |

Reviewer: ```suggestion
| due_at          | ?ISO8601 timestamp | When the reminder is due          |
```

<RouteHeader method="PUT" url="/users/@me/saved-messages/{channel.id}/{message.id}">
  Save Message
</RouteHeader>

Saves a message for the current user. Returns a [saved message](#saved-message-structure) object on success. Fires a [Saved Message Create](/topics/gateway-events#saved-message-create) Gateway event.

###### JSON Params

| Field   | Type               | Description                    |
| ------- | ------------------ | ------------------------------ |
| due_at? | ?ISO8601 timestamp | Due timestamp for the reminder |

<RouteHeader method="DELETE" url="/users/@me/saved-messages/{channel.id}/{message.id}">
  Unsave Message
</RouteHeader>

> Saves a message for the current user. Returns a empty 204 response on success. Fires a [Saved Message Delete](/topics/gateway-events#saved-message-delete) Gateway event.

Reviewer: ```suggestion
Unsaves a message for the current user. Returns a 204 empty response on success. Fires a [Saved Message Delete](/topics/gateway-events#saved-message-delete) Gateway event.
```
